tests/unit/src/commands/initialize/initDeployment.ts

```diff
@@ -0,0 +1,211 @@
+			publicKey: 'publicKey',
+			privateKey: 'privateKey',
+			encryptedKey: {
+				key: 'encryptedKey',
+				iv: 'encryptedIv'
+			}
+		});
+
+		initDeployment = loadModule('src/commands/initialize/initDeployment', {
+			'../../util/Travis': { default: TravisSpy },
+			'../../util/environment': {
+				decryptKeyName: 'decryptKey',
+				decryptIvName: 'decryptIv',
+				githubAuth: githubAuthStub,
+				keyFile: keyFileStub,
+				encryptedKeyFile: encryptedKeyFileStub
+			},
+			'./createDeployKey': { default: createDeployKeyStub },
+			'@dojo/shim/array': {
+				find: findStub
+			},
+			'fs': {
+				existsSync: existsSyncStub,
+				readFileSync: readFileSyncStub
+			}
+		});
+	},
+
+	afterEach() {
+		TravisSpy.reset();
+		githubAuthStub.reset();
+		keyFileStub.reset();
+		encryptedKeyFileStub.reset();
+		createDeployKeyStub.reset();
+		findStub.reset();
+		existsSyncStub.reset();
+		readFileSyncStub.reset();
+		isAuthorizedStub.reset();
+		travisCreateAuthorizationStub.reset();
+		travisDeleteAuthorizationStub.reset();
+		fetchRepositoryStub.reset();
+		listEnvironmentVariablesStub.reset();
+		setEnvironmentVariablesStub.reset();
+		createKeyStub.reset();
+		deleteKeyStub.reset();
+	},
+
+	'initDeployment': (() => {
+		return {
+			async 'explicit Travis instance and options'() {
+				const travis = new Travis();
+
+				await assertInitDeployment(travis, {
+					deployKeyFile: 'deploy-key.file',
+					encryptedKeyFile: 'deploy-key.enc'
+				});
+			},
+
+			async 'default instance and options; Travis is not authorized, should not create deploy key'() {
+				isAuthorizedStub.returns(false);
+				travisCreateAuthorizationStub.returns(Promise.resolve());
+
+				await assertInitDeployment();
+
+				assert.isTrue(TravisSpy.calledOnce);
+				assert.isTrue(keyFileStub.calledOnce);
+				assert.isTrue(encryptedKeyFileStub.calledOnce);
+				assert.isTrue(existsSyncStub.calledOnce);
+
+				assert.isTrue(createDeployKeyStub.notCalled);
+				assert.isTrue(createKeyStub.notCalled);
+				assert.isTrue(setEnvironmentVariablesStub.notCalled);
+
+				assert.isTrue(travisCreateAuthorizationStub.calledOnce);
+				assert.isTrue(findStub.calledOnce);
+			},
+
+			async 'should create deploy key'() {
+				listEnvironmentVariablesStub.returns(Promise.resolve([]));
+				isAuthorizedStub.returns(false);
+				existsSyncStub.returns(false);
+
+				await assertInitDeployment();
+
+				assert.isTrue(createDeployKeyStub.calledOnce);
+				assert.isTrue(readFileSyncStub.calledOnce);
+				assert.isTrue(createKeyStub.calledOnce);
+				assert.isTrue(setEnvironmentVariablesStub.calledOnce);
+			},
+
+			async 'has no key; eventually throws'() {
+				const errorMessage = 'error: cannot create key';
+
+				createKeyStub.returns(Promise.reject(errorMessage));
+
+				try {
+					await assertInitDeployment();
+				} catch (e) {
+					assert.isTrue(deleteKeyStub.calledOnce);
+					assert.strictEqual(errorMessage, e.message);
+				}
+			},
+
+			async 'has deploy key; eventually throws'() {
```

Comment:
Language updated in in 880e580. It is intentionally "has deploy key" btw -- I added an additional description though, for greater clarity with this test.